### PR TITLE
Use TimeZone from site settings in jint

### LINKED ISSUE #15501: Use time zone settings in Jint when parsing dates
### Is your feature request related to a problem? Please describe.
When parsing JS, Jint [can use a configured time zone](https://github.com/sebastienros/jint?tab=readme-ov-file#internationalization) for datetime operations. We don't do that though, so datetime operations will use something else (I don't really know, some default, perhaps?).

### Describe the solution you'd like
There's an abandoned PR for this that might be a suitable start: https://github.com/OrchardCMS/OrchardCore/pull/9650. However, it shouldn't use sync-over-async, to begin with.

### Describe alternatives you've considered
Can't add anything else.

### PR DESCRIPTION
**Pull request by jptissot**
Jint should use the TimeZone configured in the General settings when parsing dates.

Fixes #15501.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs [hunk 1]
```diff
@@ -50,9 +58,7 @@ public object Evaluate(IScriptingScope scope, string script)
                 return parser.ParseScript();
             });
 
-            var result = jsScope.Engine.Execute(parsedAst).GetCompletionValue()?.ToObject();
-
-            return result;
+            return jsScope.Engine.Evaluate(parsedAst)?.ToObject();
```
Comment (jptissot) [inline src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs:61]: This fixes the console warning: 
```
warning CS0618: 'Engine.GetCompletionValue()' is obsolete: 'Prefer calling Evaluate which returns last completion value. Execute is for initialization and Evaluate returns final result.' 
```

**Comment by Piedone** [comment]
I sincerely apologize for us taking so much time here. I checked out the PR, going over old ones, and this would be quite useful.

Is this something you'd like to revisit any time soon @jptissot?

**Comment by Piedone** [comment]
Is this something perhaps you'd take over @sebastienros?

**Comment by sebastienros** [comment]
No, we should file an issue though describing what it's trying to solve.

**Pull request closed without merging**